packages/opentelemetry-remote-sampler-aws-xray/src/remote-sampler.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Sampler, SamplingDecision, SamplingResult } from '@opentelemetry/sdk-trace-base';
+import { SamplingRule } from './remote-sampler.types';
+import axios from 'axios';
+
+const DEFAULT_INTERVAL = 5 * 60 * 1000;// 5 minutes on sampling rules fetch (default polling interval)
+
+
+// IN PROGRESS - SKELETON CLASS 
+export class AWSXRayRemoteSampler implements Sampler {
+    private _pollingInterval: number;
+    private _endpoint: string;
+    private _samplingRulesEndpoint: string;
+
+    constructor(endpoint: string, pollingInterval: number = DEFAULT_INTERVAL) {
```

Comment:
most opentelemetry components in js receive the configuration via `config` object, like [`ParentBasedSampler`](https://github.com/open-telemetry/opentelemetry-js/blob/4b04407ba34594d5febc263fe6c79fd4cf7475d5/packages/opentelemetry-sdk-trace-base/src/sampler/ParentBasedSampler.ts#L42).

This makes it easy for future addition of parameters without creating long function signatures, the retirement of deprecated parameters, and better handling of default values.

Please align the code to use a `config: AWSXRaySamplerConfig` to be consistent with the project

Reply:
I was not aware of that, thanks for calling it out! I will make those changes